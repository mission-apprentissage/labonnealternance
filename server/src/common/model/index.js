import mongoosePaginate from "mongoose-paginate"
import { getElasticInstance, mongoosastic } from "../esClient/index.js"
import schema from "../model/schema/index.js"
import { mongooseInstance } from "../mongodb.js"

const createModel = (modelName, descriptor, options = {}) => {
  const schema = new mongooseInstance.Schema(descriptor)

  if (options.esIndexName) {
    schema.plugin(mongoosastic, { esClient: getElasticInstance(), index: options.esIndexName })
  }

  if (options.esIndexName || options.paginate) {
    schema.plugin(mongoosePaginate)
  }

  if (options.createMongoDBIndexes) {
    options.createMongoDBIndexes(schema)
  }

  return mongooseInstance.model(modelName, schema)
}

const DomainesMetiers = createModel("domainesmetiers", schema.domainesMetiersSchema, {
  esIndexName: "domainesmetiers",
})
const FormationCatalogue = createModel("formationcatalogues", schema.mnaFormationSchema, {
  esIndexName: "formationcatalogues",
})
const DiplomesMetiers = createModel("diplomesmetiers", schema.diplomesMetiersSchema, {
  esIndexName: "diplomesmetiers",
})
const ApiCalls = createModel("apicalls", schema.apiCallSchema)
const Application = createModel("applications", schema.applicationSchema, {
  paginate: true,
})
const SourceFormations = createModel("sourceformations", schema.sourceFormationsSchema)
const GeoLocation = createModel("geolocation", schema.geoLocationSchema)
const EmailBlacklist = createModel("emailblacklist", schema.emailBlacklist)
const Opco = createModel("opco", schema.opco)
const BonnesBoites = createModel("bonnesboites", schema.bonneBoiteSchema, {
  esIndexName: "bonnesboites",
})
const User = createModel("user", schema.userSchema, { paginate: true })
const Appointment = createModel("appointment", schema.appointmentSchema, { paginate: true })
const WidgetParameter = createModel("widgetParameter", schema.widgetParameterSchema, { paginate: true })
const Etablissement = createModel("etablissement", schema.etablissementSchema, { paginate: true })
const ParcoursupEtablissementStat = createModel("parcoursupEtablissementStat", schema.parcoursupEtablissementStatSchema, { paginate: true })
const Formulaire = createModel("formulaire", schema.formulaireSchema, { esIndexName: "formulaires", paginate: true })
const Credential = createModel("credentials", schema.credentialSchema)
const Offre = createModel("offre", schema.offresSchema)
const Optout = createModel("optout", schema.optoutSchema, { paginate: true })
const ReferentielOpco = createModel("referentielOpco", schema.referentielOpcoSchema)
const UserRecruteur = createModel("userRecruteur", schema.userRecruteurSchema, { paginate: true })
const AppointmentDetailed = createModel("appointmentDetailed", schema.appointmentSchema)

export {
  DomainesMetiers,
  FormationCatalogue,
  DiplomesMetiers,
  ApiCalls,
  Application,
  SourceFormations,
  GeoLocation,
  EmailBlacklist,
  Opco,
  BonnesBoites,
  User,
  Appointment,
  WidgetParameter,
  Etablissement,
  ParcoursupEtablissementStat,
  Formulaire,
  Credential,
  Offre,
  Optout,
  ReferentielOpco,
  UserRecruteur,
  AppointmentDetailed,
}
