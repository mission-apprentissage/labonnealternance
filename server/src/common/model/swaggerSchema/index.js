import requireAll from "require-all"
import __dirname from "../../dirname.js"

let models = {}

function filterFile(filename) {
  if (filename.endsWith(".js") && filename !== "index.js") {
    return filename
  }

  return false
}

const modelsList = requireAll({
  dirname: __dirname(import.meta.url),
  filter: (filename) => filterFile(filename),
})

Object.keys(modelsList).forEach((filename) => {
  const model = modelsList[filename]
  models = { ...models, ...model }
})

export default models
