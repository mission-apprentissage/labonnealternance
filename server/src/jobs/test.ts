import { ObjectId } from "bson"
import { IApplicant, IApplication, ZApplicant } from "shared"

import { getDbCollection } from "@/common/utils/mongodbUtils"

import { asyncForEach } from "../common/utils/asyncUtils"

type ApplicationAggregate = Pick<IApplication, "applicant_email" | "applicant_first_name" | "applicant_last_name" | "applicant_phone" | "_id">

export const migrationTest = async () => {
  // console.log(extensions.telephone.safeParse("0690760552"))
  // return
  const applications = (await getDbCollection("applications")
    .find({}, { projection: { applicant_first_name: 1, applicant_last_name: 1, applicant_email: 1, applicant_phone: 1, _id: 1 } })
    .toArray()) as ApplicationAggregate[]

  const stat = { error: 0, success: 0, total: applications.length }

  await asyncForEach(applications, async (application) => {
    const { applicant_first_name, applicant_email, applicant_last_name, applicant_phone, _id } = application
    const now = new Date()
    const applicant: IApplicant = {
      _id: new ObjectId(),
      firstname: applicant_first_name,
      lastname: applicant_last_name,
      phone: applicant_phone,
      email: applicant_email,
      last_connection: null,
      createdAt: now,
      updatedAt: now,
    }
    const validation = ZApplicant.safeParse(applicant)
    if (!validation.success) {
      stat.error++
      console.log({ applicant, err: validation.error.errors })
    } else {
      stat.success++
    }
  })
  console.log(stat)
}
