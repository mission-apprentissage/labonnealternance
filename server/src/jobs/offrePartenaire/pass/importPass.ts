import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import rawPassModel from "shared/models/rawPass.model"

import config from "../../../config"
import { importFromUrlInXml } from "../importFromUrlInXml"
import { rawToComputedJobsPartners } from "../rawToComputedJobsPartners"

import { passJobToJobsPartners, ZPassJob } from "./passMapper"

const rawCollectionName = rawPassModel.collectionName
const offerXmlTag = "item"

export const importPassRaw = async () => await importFromUrlInXml({ destinationCollection: rawCollectionName, url: config.passUrl, offerXmlTag })

export const importPassToComputed = async () => {
  await rawToComputedJobsPartners({
    collectionSource: rawCollectionName,
    partnerLabel: JOBPARTNERS_LABEL.HELLOWORK,
    zodInput: ZPassJob,
    mapper: passJobToJobsPartners,
    documentJobRoot: offerXmlTag,
  })
}
