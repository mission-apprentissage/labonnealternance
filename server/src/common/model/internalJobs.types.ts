import { ObjectId } from "mongodb"

import { CronName } from "@/jobs/jobs"

/*
  name: Le nom de la tâche
  type: Type du job simple ou cron
  status: Statut courant du job
  sync: Si le job est synchrone
  payload: La donnée liéé à la tâche
  output: Les valeurs de retours du job
  cron_string: standard cron string exemple: '*\/2 * * * *'
  scheduled_for: Date de lancement programmée
  started_at: Date de lancement
  ended_at: Date de fin d'execution
  updated_at: Date de mise à jour en base de données
  created_at: Date d'ajout en base de données
*/

interface IInternalJobsSimple {
  _id: ObjectId
  name: string
  type: "simple"
  status: "pending" | "will_start" | "running" | "finished" | "blocked" | "errored"
  sync: boolean
  payload: Record<string, any>
  output?: Record<string, unknown> | undefined
  scheduled_for: Date
  started_at?: Date
  ended_at?: Date
  updated_at: Date
  created_at: Date
}

interface IInternalJobsCron {
  _id: ObjectId
  name: CronName
  type: "cron"
  status: "pending" | "will_start" | "running" | "finished" | "blocked" | "errored"
  sync: boolean
  cron_string: string
  scheduled_for: Date
  updated_at: Date
  created_at: Date
}

interface IInternalJobsCronTask {
  _id: ObjectId
  name: CronName
  type: "cron_task"
  status: "pending" | "will_start" | "running" | "finished" | "blocked" | "errored"
  sync: boolean
  scheduled_for: Date
  started_at?: Date
  ended_at?: Date
  updated_at: Date
  created_at: Date
}

type IInternalJobs = IInternalJobsSimple | IInternalJobsCron | IInternalJobsCronTask

export type { CronName, IInternalJobs, IInternalJobsCron, IInternalJobsCronTask, IInternalJobsSimple }
