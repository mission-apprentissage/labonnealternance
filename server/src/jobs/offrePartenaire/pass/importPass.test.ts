import fs from "node:fs"

import { JOBPARTNERS_LABEL } from "shared/models/jobsPartners.model"
import { beforeEach, describe, expect, it, vi } from "vitest"

import { getDbCollection } from "@/common/utils/mongodbUtils"
import { useMongo } from "@tests/utils/mongo.test.utils"

import { importPassRaw, importPassToComputed } from "./importPass"

const now = new Date("2024-07-21T04:49:06.000+02:00")

describe("importPass", () => {
  useMongo()

  beforeEach(() => {
    vi.useFakeTimers()
    vi.setSystemTime(now)

    return async () => {
      vi.useRealTimers()
      await getDbCollection("computed_jobs_partners").deleteMany({})
      await getDbCollection("raw_pass").deleteMany({})
    }
  })

  it("should test the import of pass data into computed_job_partners", async () => {
    const fileStream = fs.createReadStream("server/src/jobs/offrePartenaire/pass/importPass.test.input.rss")
    await importPassRaw(fileStream)
    expect.soft(await getDbCollection("raw_pass").countDocuments({})).toBe(5)

    await importPassToComputed()
    const jobs = (
      await getDbCollection("computed_jobs_partners")
        .find({ partner_label: JOBPARTNERS_LABEL.PASS }, { projection: { _id: 0, created_at: 0 } })
        .toArray()
    ).sort((a, b) => ((a.partner_job_id ?? "") < (b.partner_job_id ?? "") ? -1 : 1))
    expect.soft(jobs.length).toBe(5)
    expect.soft(jobs).toMatchSnapshot()
  })
})
