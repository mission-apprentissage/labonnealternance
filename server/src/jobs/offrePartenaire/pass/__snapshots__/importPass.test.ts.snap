// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`importPass > should test the import of pass data into computed_job_partners 1`] = `
[
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": "0666666666",
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/chargee-de-projet-formationcommunication",
    "business_error": null,
    "contract_duration": 24,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-09-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T13:43:07.000Z,
    "offer_description": "<p>Dans le cadre de la transformation numérique des formations au sein de la DSNA, nous recherchons deux alternants disponibles à partir de septembre 2025 pour le projet de création de supports numériques d’information et de formation (enseignement assisté par ordinateur) pour la formation continue des contrôleurs aériens.</p><p>Les alternants seront amenés à réaliser les missions suivantes :</p><ul><li>Communiquer avec les enseignants et comprendre les objectifs pédagogiques des formations à réaliser</li><li>Accompagner la transformation de documents en storyboards et travailler au design des formations résultantes à l’aide du logiciel Articulate Storyline</li><li>Concevoir et réaliser des media numériques (images/vidéos/audio) en s’appuyant, en tant que de besoin, sur des outils intégrant l’intelligence artificielle (Vyond, Murf)</li><li>Présenter, organiser des démonstrations du logiciel Storyline à des fins de promotion du logiciel Articulate au sein de la DSNA et répondre aux questions des utilisateurs&nbsp;</li><li>Répondre aux besoins des projets en termes de communication et de marketing : création de visuels, évaluation des retours d’expériences</li></ul><p>Concevoir, mettre en œuvre et évaluer des projets de communication interne et/ou externe</p>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations niveau (Bac+3 à Bac+4)",
    },
    "offer_title": "Chargé(e) de projet - Formation/Communication",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-158984",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "ATHIS MONS 91",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "Direction Générale de l&#039;Aviation Civile",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": "0666666666",
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/chargee-de-projet-formationcommunication-0",
    "business_error": null,
    "contract_duration": 24,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-09-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T13:57:19.000Z,
    "offer_description": "<p>Dans le cadre de la transformation numérique des formations au sein de la DSNA, nous recherchons deux alternants disponibles à partir de septembre 2025 pour le projet de création de supports numériques d’information et de formation (enseignement assisté par ordinateur) pour la formation continue des contrôleurs aériens.</p><p>Les alternants seront amenés à réaliser les missions suivantes :</p><ul><li>Communiquer avec les enseignants et comprendre les objectifs pédagogiques des formations à réaliser</li><li>Accompagner la transformation de documents en storyboards et travailler au design des formations résultantes à l’aide du logiciel Articulate Storyline</li><li>Concevoir et réaliser des media numériques (images/vidéos/audio) en s’appuyant, en tant que de besoin, sur des outils intégrant l’intelligence artificielle (Vyond, Murf)</li><li>Présenter, organiser des démonstrations du logiciel Storyline à des fins de promotion du logiciel Articulate au sein de la DSNA et répondre aux questions des utilisateurs&nbsp;</li><li>Répondre aux besoins des projets en termes de communication et de marketing : création de visuels, évaluation des retours d’expériences</li><li>Concevoir, mettre en œuvre et évaluer des projets de communication interne et/ou externe</li></ul>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations niveau (Bac+3 à Bac+4)",
    },
    "offer_title": "Chargé(e) de projet - Formation/Communication",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-158987",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "TOULOUSE 31",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "Direction Générale de l&#039;Aviation Civile",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": null,
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/contrat-dapprentissage-education-aux-images-hf",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-04-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T13:47:42.000Z,
    "offer_description": "<p>Sous la responsabilité de la chargée de mission en charge du suivi du dispositif Ma Classe au cinéma, le service des publics recrute un(e) apprenti(e) chargé(e) de participer au projet de création d’un site de l’éducation aux images porté par le CNC.</p><p>Ce site s’adressera à tous les acteurs de l’éducation aux images et aux enseignants. Il permettra de valoriser la politique d’éducation aux images du Centre et de mettre en avant les ressources pédagogiques Collège au cinéma et Lycéens et Apprentis au cinéma produites par le CNC, qui travaille à leur dématérialisation et leur actualisation.</p><p>&nbsp;</p><p><strong>Principales missions:</strong></p><ul><li>Participation au groupe-projet de développement du site</li><li>Recensement des ressources existantes autour des dispositifs Collège au cinéma et Lycéens et Apprentis au cinéma, notamment les ressources &nbsp; &nbsp; pédagogiques (ressources écrites, vidéo, iconographiques),</li><li><p>Suivi des travaux de réécriture et de traduction numérique des dossiers enseignants et fiches élèves Collège au cinéma et Lycéens et Apprentis au &nbsp; &nbsp;cinéma</p><p>Recueil et recensement des ressources propres aux autres actions d’éducation aux images portées ou accompagnées par le service (Mission de &nbsp;sensibilisation à l’écriture scénaristique / défi Ecris ta série&nbsp;!, programme des Ambassadeurs jeunes du cinéma, Opération Des Cinés, La Vie&nbsp;!, actions de &nbsp;diffusion culturelle et de développement des publics…)</p><p>Travail ponctuel sur d’autres missions relatives au service des publics, notamment le suivi des comités de sélection des films Ma classe au cinéma.</p></li></ul>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-06-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "7",
      "label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    },
    "offer_title": "Contrat d'apprentissage - éducation aux images H/F",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-158990",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "PARIS 14ème arrondissement 75",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "Centre national du cinéma et de l&#039;image animée",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": "04.66.66.66.66",
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/apprentie-redacteur-en-matiere-domanial-equivalent-attache-cat-division-des-domaine-service-4",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-09-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T14:08:58.000Z,
    "offer_description": "<p>La direction départementale des finances publiques (DDFiP) et son réseau territorial est en charge des missions fiscales (établissement des impositions des personnes physiques et morales, encaissement des impôts, contrôle fiscal et contentieux) et de gestion publique (tenue des comptes de l’État et des collectivités locales, paiement des dépenses de l’État, Domaine).</p><p>Au siège départemental, s'exercent plus particulièrement des fonctions de pilotage et d’animation des services locaux, de soutien à l’activité (ressources humaines, budget, immobilier), ainsi que la mission domaniale.</p><p>L’activité du Domaine consiste à gérer le patrimoine immobilier de l’État et des différentes implantations des services de l’État sur le département. Le Domaine est également chargé des travaux d'évaluation qui encadrent les opérations de gestion précitées mais aussi les opérations immobilières des collectivités locales. Le Domaine traite les dossiers contentieux de l’expropriation en assurant la fonction de Commissaire du gouvernement auprès du juge de l’expropriation. Enfin, le Domaine est également investi d'une mission d'intérêt général de gestion des patrimoines privés recueillis à l'occasion des successions vacantes (successions non réclamées, sans héritiers ou refusées).</p><p>Le candidat intègre le Service de gestion des patrimoines privés au sein de la Division Domaine de la DDFiP 06.&nbsp;</p><p>Ce service est en charge la gestion des patrimoines privés recueillis à l'occasion des successions vacantes (successions non réclamées, sans héritiers ou refusées).&nbsp;</p><p>Cette mission consiste d’une part à récupérer tous les actifs de la succession, d’autre part à recenser les dettes afin que celles-ci puissent être payées à concurrence des actifs disponibles.&nbsp;</p><p>Dans le cadre de la gestion des intérêts de la succession, l’apprenti au sein du service de gestion du patrimoine privé aura en charge le traitement des dossiers&nbsp;: recensement de l’actif et du passif successoral.</p><p>Il s’agira également de procéder, en relation avec nos partenaires et dans les meilleures conditions, aux cessions de biens immobiliers.</p><p>Enfin, l’apprenti(e) pourra être chargé(e) d'analyser les éventuelles revendications tardives des héritiers.</p>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations niveau (Bac+3 à Bac+4)",
    },
    "offer_title": "APPRENTI(E) - Rédacteur en matière domanial - équivalent attaché (cat. A) Division des Domaine – Service Gestion des patrimoines privés",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-158999",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "NICE 6",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "DGFIP",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": "04.66.66.66.66",
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/apprentie-redacteur-en-matiere-domanial-equivalent-attache-cat-division-du-domaine-service-6",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-09-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T14:09:12.000Z,
    "offer_description": "<p>La direction départementale des finances publiques (DDFiP) et son réseau territorial est en charge des missions fiscales (établissement des impositions des personnes physiques et morales, encaissement des impôts, contrôle fiscal et contentieux) et de gestion publique (tenue des comptes de l’État et des collectivités locales, paiement des dépenses de l’État, Domaine).&nbsp;</p><p>Au siège départemental, s'exercent plus particulièrement des fonctions de pilotage et d’animation des services locaux, de soutien à l’activité (ressources humaines, budget, immobilier), ainsi que la mission domaniale.</p><p>L’activité du Domaine consiste à gérer le patrimoine immobilier de l’État et des différentes implantations des services de l’État sur le département. Le Domaine est également chargé des travaux d'évaluation qui encadrent les opérations de gestion précitées mais aussi les opérations immobilières des collectivités locales. Le Domaine traite les dossiers contentieux de l’expropriation en assurant la fonction de Commissaire du gouvernement auprès du juge de l’expropriation. Enfin, le Domaine est également investi d'une mission d'intérêt général de gestion des patrimoines privés recueillis à l'occasion des successions vacantes (successions non réclamées, sans héritiers ou refusées).&nbsp;</p><p>Le candidat intègre le Service Local du Domaine de la DDFIP 06, ce service de Gestion a pour objet de valoriser le patrimoine de l’État (location, vente et facturation des occupations).&nbsp;</p><p>L’apprenti sera en charge d’analyser et de traiter les dossiers d’occupation du domaine public afin d’en déterminer la redevance. Des missions ponctuelles seront également (i.e. du traitement de servitudes, transfert de gestion, etc.)</p>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations niveau (Bac+3 à Bac+4)",
    },
    "offer_title": "APPRENTI(E) - Rédacteur en matière domanial - équivalent attaché (cat. A) Division du Domaine – Service Local du Domaine",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-159002",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "NICE 6",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "DGFIP",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": "04.66.66.66.66",
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/apprentie-redacteur-en-matiere-juridique-et-fiscal-equivalent-attache-cat-pole-de-controle",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-09-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T14:09:41.000Z,
    "offer_description": "<p>Au sein du ministère de l’Économie, des Finances et de la Souveraineté industrielle et numérique, la Direction générale des Finances publiques (DGFiP) est une administration au cœur du fonctionnement de l’État. Grâce à leur expertise, leur savoir-être et leur engagement, les agents des Finances publiques contribuent à faciliter toujours davantage le quotidien de l’ensemble des usagers et à faire respecter l’égalité de tous devant les charges publiques et à collecter les impôts qui viendront financer la vie publique. Administration performante et innovante, notre transformation informatique et numérique (big data,intelligence artificielle, cloud…) place l’utilisateur au centre de nos démarches.</p><p>Dans le département des Alpes-Maritimes, la direction départementale des finances publiques (DDFIP 06) pilote et anime la mission du contrôle fiscale mise en œuvre par les structures infra-départementales sur le périmètre du département.</p><p>Les Pôle de contrôles revenus/patrimoine (PCRP) participent au contrôle fiscal des particuliers en lien avec la Division du contrôle fiscal en Direction et les services des impôts des particuliers. Les PCRP ont pour mission principale de contrôler les dossiers des particuliers sur tous les domaines de la fiscalité (impôt sur le revenu, IFI, donations, successions) et en examinant la cohérence entre les déclarations de revenus et de patrimoine.</p><p>Au sein du PCRP de Cannes, le candidat participera aux différentes missions&nbsp;:</p><p>- Contrôle sur pièces des dossiers patrimoniaux et rédaction des pièces de procédure &nbsp;;</p><p>- Instruction des contentieux d’assiette des dossiers patrimoniaux&nbsp;;</p><p>- Traitement des dossiers d’accréditation en matière de plus-values immobilières.</p>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "6",
      "label": "Licence, Maîtrise, autres formations niveau (Bac+3 à Bac+4)",
    },
    "offer_title": "APPRENTI(E) - Rédacteur en matière juridique et fiscal -  équivalent attaché (cat. A) Pôle de contrôle revenus patrimoine de Cannes",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-159005",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "NICE 6",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "DGFIP",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
  {
    "apply_email": "mail@mail.fr",
    "apply_phone": null,
    "apply_url": "https://www.pass.fonction-publique.gouv.fr/offre/contrat-dapprentissage-financements-hf",
    "business_error": null,
    "contract_duration": 12,
    "contract_is_disabled_elligible": false,
    "contract_remote": null,
    "contract_start": 2025-09-14T20:00:00.000Z,
    "contract_type": [
      "Apprentissage",
    ],
    "errors": [],
    "jobs_in_success": [],
    "offer_access_conditions": [],
    "offer_creation": 2025-02-26T14:40:27.000Z,
    "offer_description": "<p>Au sein du service, l’alternant(e) agira notamment en appui sur les missions suivantes&nbsp;:</p><ul><li>la veille et l’étude des outils de financement du secteur</li><li>la tutelle des SOFICA et des crédits d’impôt ;</li><li>le suivi des opérateurs du financement, dont Bpifrance et l’IFCIC</li><li>le suivi des nouvelles initiatives de financements privés et des stratégies de développement des entreprises</li><li>la préparation des projets de loi de finances sur les sujets qui concernent le service</li></ul>",
    "offer_desired_skills": [],
    "offer_expiration": 2025-11-14T20:00:00.000Z,
    "offer_multicast": true,
    "offer_opening_count": 1,
    "offer_origin": null,
    "offer_rome_codes": null,
    "offer_status": "Active",
    "offer_status_history": [],
    "offer_target_diploma": {
      "european": "7",
      "label": "Master, titre ingénieur, autres formations niveau (Bac+5)",
    },
    "offer_title": "Contrat d'Apprentissage - financements H/F",
    "offer_to_be_acquired_knowledge": [],
    "offer_to_be_acquired_skills": [],
    "partner_job_id": "A-2025-159026",
    "partner_label": "PASS",
    "updated_at": 2024-07-21T02:49:06.000Z,
    "validated": false,
    "workplace_address_city": null,
    "workplace_address_label": "PARIS 14ème arrondissement 75",
    "workplace_address_street_label": null,
    "workplace_address_zipcode": null,
    "workplace_brand": null,
    "workplace_description": null,
    "workplace_geopoint": null,
    "workplace_idcc": null,
    "workplace_legal_name": null,
    "workplace_naf_code": null,
    "workplace_naf_label": null,
    "workplace_name": "Centre national du cinéma et de l&#039;image animée",
    "workplace_opco": null,
    "workplace_siret": null,
    "workplace_size": null,
    "workplace_website": null,
  },
]
`;
